### PR TITLE
feat: Create new Date parameter. Created tests. Updated swagger.

### PR DESCRIPTION
**Pull request by krackerman**
#754 added support for Date Param

### CODE REVIEW

File: src/params/date.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+const moment = require('moment');
+const RegEx = require('./regex');
+
+class Date extends RegEx {
+  constructor(name, position, opts) {
+    super(name, position, Object.assign({}, opts, {
+      regex: new RegExp(/^(\d{4})-(\d{2})-(\d{2})$/)
```
Comment (simlu) [inline src/params/date.js]: This should be more restrictive

Comment (MacMcIrish) [reply src/params/date.js]: Why use capture groups?

File: src/params/date.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    }));
+  }
+
+  validate(value) {
+    let valid = super.validate(value);
+    if (valid && !moment(value).isValid()) {
```
Comment (MacMcIrish) [inline src/params/date.js]: What about if `value === undefined`? Is that a real possibility here?

Comment (MacMcIrish) [reply src/params/date.js]:
```suggestion
    if (typeof value !== 'string') {
      valid = false;
    }
    if (valid && !moment(value).isValid()) {
```

File: src/params/date.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
+const moment = require('moment');
+const RegEx = require('./regex');
+
+class Date extends RegEx {
+  constructor(name, position, opts) {
+    super(name, position, Object.assign({}, opts, {
+      regex: new RegExp(/^\d{4}-\d{2}-\d{2}$/)
```
Comment (MacMcIrish) [inline src/params/date.js]: Still needs to be more restrictive

**Pull request merged**